flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/internal/MongoDBConnectorSourceTask.java

```diff
@@ -0,0 +1,229 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.mongodb.internal;
+
+import com.mongodb.kafka.connect.source.MongoSourceTask;
+import io.debezium.connector.AbstractSourceInfo;
+import io.debezium.connector.SnapshotRecord;
+import io.debezium.data.Envelope;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+import org.apache.kafka.connect.source.SourceTaskContext;
+import org.bson.json.JsonReader;
+
+import java.lang.reflect.Field;
+import java.time.Instant;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Source Task that proxies mongodb kafka connector's {@link MongoSourceTask} to adapt to {@link
+ * com.ververica.cdc.debezium.internal.DebeziumChangeFetcher}.
+ */
+public class MongoDBConnectorSourceTask extends SourceTask {
+
+    private static final String TRUE = "true";
+
+    private static final Schema HEARTBEAT_VALUE_SCHEMA =
+            SchemaBuilder.struct()
+                    .field(AbstractSourceInfo.TIMESTAMP_KEY, Schema.INT64_SCHEMA)
+                    .build();
+
+    private final MongoSourceTask target;
+
+    private final Field isCopyingField;
+
+    private SourceRecord currentLastSnapshotRecord;
+
+    private boolean isInSnapshotPhase = false;
+
+    public MongoDBConnectorSourceTask() throws NoSuchFieldException {
+        this.target = new MongoSourceTask();
+        this.isCopyingField = MongoSourceTask.class.getDeclaredField("isCopying");
+    }
+
+    @Override
+    public String version() {
+        return target.version();
+    }
+
+    @Override
+    public void initialize(SourceTaskContext context) {
+        this.target.initialize(context);
+        this.context = context;
+    }
+
+    @Override
+    public void start(Map<String, String> props) {
+        target.start(props);
+        isInSnapshotPhase = isCopying();
+    }
+
+    @Override
+    public void commit() throws InterruptedException {
+        target.commit();
+    }
+
+    @Override
+    public void commitRecord(SourceRecord record) throws InterruptedException {
+        target.commitRecord(record);
+    }
+
+    @Override
+    public void commitRecord(SourceRecord record, RecordMetadata metadata)
+            throws InterruptedException {
+        target.commitRecord(record, metadata);
+    }
+
+    @Override
+    public List<SourceRecord> poll() throws InterruptedException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.